cpp/cmake/Modules/FindcuFile.cmake

```diff
@@ -93,6 +93,12 @@ find_package_handle_standard_args(cuFile
     cuFile_VERSION
 )
 
+if (cuFile_FOUND AND NOT TARGET cuFile::cuFile_interface)
+  add_library(cuFile::cuFile_interface IMPORTED INTERFACE)
+  target_include_directories(cuFile::cuFile_interface INTERFACE "$<BUILD_INTERFACE:${cuFile_INCLUDE_DIR}>")
+  target_compile_options(cuFile::cuFile_interface INTERFACE "${cuFile_COMPILE_OPTIONS}")
+  target_compile_definitions(cuFile::cuFile_interface INTERFACE CUFILE_FOUND)
+endif ()
 
 if (cuFile_FOUND AND NOT TARGET cuFile::cuFile)
```

Comment:
I think we're going to need to handle the situation that we're creating in the build containers where we only have the header available, but don't have the shared library available. This is going to set `cuFile_FOUND` to false since `cuFile_LIBRARY` and `cuFileRDMA_LIBRARY` will both be `NOTFOUND`.

Reply:
Should we use `cuFile_INCLUDE_DIR` in this condition (and in the main CMake file), then?

Reply:
No I think we should just change this from `cuFile_FOUND` --> `cuFile_INCLUDE_DIR` at line 96

Reply:
wouldn't we need to change line 
https://github.com/rapidsai/cudf/pull/7444/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.hpp&file-filters%5B%5D=.txt#diff-1bba462ab050e89360fd88110a689e85ee037749cea091a1848ab574381d3795R472
in order to "link" to the interface cuFile target, since cuFile_FOUND would still be false when only the header is present?

Reply:
Yes, in that line we could use something like:
```
if(TARGET cuFile::cuFile_interface)
```

Reply:
I believe that when `cuFile_LIBRARY` is not found we are going to error out on `find_package_handle_standard_args` since we have marked that variable as being required for the package to be `found`.